packages/node/src/configure/configure.module.ts

```diff
@@ -112,12 +116,17 @@ export class ConfigureModule {
       setLevel('debug');
     }
 
-    const projectPath = config.ipfs
-      ? argv.subquery
-      : path.resolve(
-          config.configDir && !argv.subquery ? config.configDir : '.',
-          config.subquery,
-        );
+    let projectPath: string;
+
+    const ipfsMatch = argv.subquery.match(IPFS_REGEX);
+    if (ipfsMatch) {
+      projectPath = argv.subquery;
+    } else {
+      projectPath = path.resolve(
+        config.configDir && !argv.subquery ? config.configDir : '.',
```

Comment:
The origin implementation seems too complicated. argv.subquery should be mandatory then here can be simplified to `projectPath = argv.subquery`, `path.resolve()` can be done in localreader.